test/test_option.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2020 Canonical, Ltd.
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the launch api."""
+
+import pytest
+
+from ros2launch_security.option.security import _Keystore
+from ros2launch_security.option.security import InvalidKeystoreError
+from ros2launch_security.option.security import NoKeystoreProvidedError
+from ros2launch_security.option.security import NonexistantKeystoreError
+
+import sros2.api._keystore
+
+
+@pytest.fixture
+def keystore_path(tmp_path):
+    sros2.api._keystore.create_keystore(tmp_path)
+    return tmp_path
+
+
+def test__keystore_existing_keystore(keystore_path):
+    res = _Keystore(keystore_path=keystore_path, create_keystore=False)
+    assert sros2.api._keystore.is_valid_keystore(keystore_path)
+
+    # Test that it doesn't delete a provided keystore when done
+    del res
```

Comment:
@gbiggs invoking `del` on a variable simply decrements its ref count. You may want to force garbage collection for this test to test what you want it to test :) Same elsewhere.

Reply:
Fixed in db0237e.